### PR TITLE
Fix base image pinning during upgrades via cluster/gce/upgrade.sh

### LINKED ISSUE #33056: Upgrade 1.3.7 w/ GCI -> 1.4.0-beta.8 failed: health-monitor.sh not found
Kubelet not running after upgrading GCI nodes to `1.4.0-beta.8`:

1. Create cluster running v1.3.7 on GCI 54 (see [config](https://gist.github.com/timstclair/b8ec1867338073bc0ed62ea4bcf1ffa3#file-patch-diff))
2. Upgrade masters to 1.4.0-beta.8 `[(v1.4.0-beta.8)] $ cluster/gce/upgrade.sh -M v1.4.0-beta.8`
3. Upgrade nodes to 1.4.0-beta.8 `[(v1.4.0-beta.8)] $ cluster/gce/upgrade.sh -N v1.4.0-beta.8`

```
...
Rolling update 69a6b7dd-2bea-4f95-8959-e8857b1bdd07 is ROLLED_OUT state and finished.
Deleted [https://www.googleapis.com/compute/v1/projects/stclair-k8s-ext/global/instanceTemplates/kubernetes-minion-template].
== Finished upgrading nodes to v1.4.0-beta.8. ==
== Validating cluster post-upgrade ==
Waiting for 4 ready nodes. 1 ready nodes, 4 registered. Retrying.
...
Detected 1 ready nodes, found 4 nodes out of expected 4. Your cluster may not be fully functional.
NAME                           STATUS                     AGE
kubernetes-master              Ready,SchedulingDisabled   34m
kubernetes-minion-group-3qca   NotReady                   2d
kubernetes-minion-group-nmd2   NotReady                   2d
kubernetes-minion-group-vshq   NotReady                   2d
```
On a node:
```
$ sudo journalctl -a | grep kube | tail -n 20 
Sep 19 23:39:23 kubernetes-minion-group-3qca systemd[393]: kubelet-monitor.service: Service hold-off time over, scheduling restart.
Sep 19 23:39:23 kubernetes-minion-group-3qca systemd[393]: kube-docker-monitor.service: Service hold-off time over, scheduling restart.
Sep 19 23:39:23 kubernetes-minion-group-3qca systemd[393]: Stopped Kubernetes health monitoring for docker.
Sep 19 23:39:23 kubernetes-minion-group-3qca systemd[393]: Starting Kubernetes health monitoring for docker...
Sep 19 23:39:23 kubernetes-minion-group-3qca systemd[393]: Stopped Kubernetes health monitoring for kubelet.
Sep 19 23:39:23 kubernetes-minion-group-3qca chmod[18211]: chmod: cannot access '/home/kubernetes/bin/health-monitor.sh': No such file or directory
$ ls /home/kubernetes/bin
bridge  cnitool  configure.sh  dhcp  flannel  host-local  ipvlan  kubectl  kubelet  loopback  macvlan  noop  ptp  tuning
$ systemctl | grep kubelet
  kubelet-monitor.service                                                                      loaded activating auto-restart Kubernetes health monitoring for kubelet
```

No failures in `sudo journalctl -u cloud*`.

/cc @vishh @kubernetes/goog-image @davidopp

### PR DESCRIPTION
**Pull request by vishh**
Fixes #33056

cc @timstclair @maisem

``` release-note
Fix base image pinning during upgrades via cluster/gce/upgrade.sh
```

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33147)

### CODE REVIEW

**Comment by vishh** [comment]
cc @Amey-D

**Comment by timstclair** [comment]
I'm struggling to trace this code. Can you summarize what the issue was, and how this fixes it?

**Comment by vishh** [comment]
This PR ensures that we set the KUBE_NODE_OS_DISTRIBUTION env variable
early enough for the rest of the upgrade logic to setup the env variables
for the upgrade appropriately.
This PR still doesn't fix gci -> gci upgrades fully. I will appreciate any
help in fixing that upgrade path completely.

**Comment by timstclair** [comment]
Got it. LGTM.

**Comment by mtaufen** [comment]
Jenkins GCI GKE smoke e2e failed with this near the end of the log:
```
ERROR: (gcloud.container.clusters.create) Operation [<Operation
 name: u'operation-1474670221656-ee661d04'
 operationType: OperationTypeValueValuesEnum(CREATE_CLUSTER, 1)
 selfLink: u'https://test-container.sandbox.googleapis.com/v1/projects/943316867058/zones/us-central1-f/operations/operation-1474670221656-ee661d04'
 status: StatusValueValuesEnum(DONE, 3)
 statusMessage: u'Timed out waiting for cluster initialization. Cluster API may not be available.'
 targetLink: u'https://test-container.sandbox.googleapis.com/v1/projects/943316867058/zones/us-central1-f/clusters/e2e-gke-agent-pr-66-0'
 zone: u'us-central1-f'>] finished with error: Timed out waiting for cluster initialization. Cluster API may not be available.
```

**Comment by mtaufen** [comment]
@k8s-bot gci gke test this issue #IGNORE

**Comment by mtaufen** [comment]
@jessfraz This fix should be cherrypicked into 1.4.

**Comment by jessfraz** [comment]
cherry-picked from #34628
